stdlib/src/tempfile/tempfile.mojo

```diff
@@ -164,3 +164,84 @@ fn mkdtemp(
         except:
             continue
     raise Error("Failed to create temporary file")
+
+
+# TODO use shutil.rmtree (or equivalent) when it exists
+fn _rmtree(path: String, ignore_errors: Bool = False) raises:
```

Comment:
Can we use `PathLike` for the `path` argument?

Reply:
`listdir` returns a list of strings and `os.path.join` only takes `String` for now, so found it easier to use a `String` for now

Reply:
Okay, thanks! Would it be possible to fix this in a followup?